packages/google_maps_flutter/lib/src/tile_overlay.dart

```diff
@@ -79,7 +79,7 @@ class TileOverlay {
 
   /// The tile overlay's zIndex, i.e., the order in which it will be drawn where
   /// overlays with larger values are drawn above those with lower values
-  final double zIndex;
+  final int zIndex;
```

Comment:
This param have to be a double. Because Android plugin waiting float value

Reply:
On the Android SDK the zIndex is always a `float` value (`double` in dart) but on the iOS SDK an `int` is required. 

Since all the other type of overlays (like polygon, circle, polyline) are using the type `int` instead of double, except for TileOverlay, I unified this one to be `int`.  

Sources:
[Android Polyline](https://developers.google.com/android/reference/com/google/android/gms/maps/model/Polyline.html#getZIndex())
[iOS Polyline](https://developers.google.com/maps/documentation/ios-sdk/reference/interface_g_m_s_overlay.html#a0e9871b40c30bb39f9a45dadaf7a0f75)
[Flutter Polyline](https://github.com/flutter/plugins/blob/master/packages/google_maps_flutter/lib/src/polyline.dart#L125)

Reply:
@neokree Ok yes. You are right